### PR TITLE
Image Data Generators

### PR DESCRIPTION
**Pull request by jsalbert**
Added 2 different Keras Image Data Generators:

- `BalancedImageDataGenerator`: generates batches with equally probable class sampling. Note that there could be repetition of samples in the same epoch. 

- `ImageDataGeneratorSameMultiGT`:  Has as parameter `n_outputs`, every batch contains `n_outputs` times n_outputs times the ground truth labels. This is used for multi-loss functions.

### CODE REVIEW

**Comment by jsalbert** [comment]
@anirudhcoder next version will include it! 🚢 💯
